### PR TITLE
fixes gravatar fallback for team collective events

### LINKED ISSUE #1757: Team member doesn't show up in team public page if their avatar doesn't exist
Found a bug? Please fill out the sections below. 👍

### Issue Summary

When the user doesn't have an avatar uploaded, they can still be added to a team event but they don't show up on the public page of the event, or on the listing of the event

### Steps to Reproduce

1. Add a team member who doesn't have their avatar uploaded, to a team event
2. Check event listing and look for their gravatar fallback
3. Check the public page and look for their gravatar fallback

[reported by @PeerRich ]

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Integrates `[user]/avatar.png` into team collective events

Fixes #1757 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Visit team collective event setting in `/event-types/[type]`, `/event-types/` and `/team/[slug]/[type]` and the user without avatar should now be visible with their gravatar.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -126,7 +126,11 @@ const AvailabilityPage = ({ profile, eventType, workingHours }: Props) => {
                           .filter((user) => user.name !== profile.name)
                           .map((user) => ({
                             title: user.name,
-                            image: user.avatar || undefined,
+                            image:
+                              (process.env.NEXT_PUBLIC_APP_URL || process.env.BASE_URL) +
+                              "/" +
+                              user.username +
+                              "/avatar.png",
```
Comment (zomars) [inline components/booking/pages/AvailabilityPage.tsx:133]: `process.env.BASE_URL` will always be undefined in the frontend.

```suggestion
                            image: `${process.env.NEXT_PUBLIC_APP_URL}/${user.username}/avatar.png`,
```

File: components/booking/pages/AvailabilityPage.tsx [hunk 2]
```diff
@@ -175,7 +179,11 @@ const AvailabilityPage = ({ profile, eventType, workingHours }: Props) => {
                           .map((user) => ({
                             title: user.name,
                             alt: user.name,
-                            image: user.avatar,
+                            image:
+                              (process.env.NEXT_PUBLIC_APP_URL || process.env.BASE_URL) +
+                              "/" +
+                              user.username +
+                              "/avatar.png",
```
Comment (zomars) [inline components/booking/pages/AvailabilityPage.tsx:186]:
```suggestion
                            image: `${process.env.NEXT_PUBLIC_APP_URL}/${user.username}/avatar.png`,
```

File: pages/event-types/[type].tsx [hunk 3]
```diff
@@ -313,15 +312,15 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
   }: {
     id: number | null;
     name: string | null;
-    avatar: string | null;
+    username: string | null;
   }) => ({
     value: `${id || ""}`,
     label: `${name || ""}`,
-    avatar: `${avatar || ""}`,
+    avatar: `${(process.env.NEXT_PUBLIC_APP_URL || process.env.BASE_URL) + "/" + username + "/avatar.png"}`,
```
Comment (zomars) [inline pages/event-types/[type].tsx:323]:
```suggestion
    avatar: `${process.env.NEXT_PUBLIC_APP_URL}/${username}/avatar.png`,
```

File: pages/event-types/[type].tsx [hunk 4]
```diff
@@ -1712,7 +1711,8 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
   const teamMembers = eventTypeObject.team
     ? eventTypeObject.team.members.map((member) => {
         const user = member.user;
-        user.avatar = user.avatar || defaultAvatarSrc({ email: asStringOrUndefined(user.email) });
+        user.avatar =
+          (process.env.NEXT_PUBLIC_APP_URL || process.env.BASE_URL) + "/" + user.username + "/avatar.png";
```
Comment (zomars) [inline pages/event-types/[type].tsx:1715]:
```suggestion
        user.avatar = `${process.env.NEXT_PUBLIC_APP_URL}/${user.username}/avatar.png`;
```

File: pages/event-types/index.tsx [hunk 5]
```diff
@@ -153,7 +153,11 @@ const EventTypeList = ({ readOnly, types, profile }: EventTypeListProps): JSX.El
                         truncateAfter={4}
                         items={type.users.map((organizer) => ({
                           alt: organizer.name || "",
-                          image: organizer.avatar || "",
+                          image:
+                            (process.env.NEXT_PUBLIC_APP_URL || process.env.BASE_URL) +
+                            "/" +
+                            organizer.username +
```
Comment (zomars) [inline pages/event-types/index.tsx:159]: We have a type error here:

> Property 'username' does not exist on type '{ name: string | null; id: number; avatar: string | null; }'.ts(2339)

**Review by zomars — CHANGES_REQUESTED** [review]
Thank you for doing this, let's just address some comments first.

File: pages/event-types/[type].tsx (hunk 4 above)
Comment (alishaz-polymath) [reply pages/event-types/[type].tsx:1719]: Yep, got it. Fixing now! :)

File: pages/event-types/index.tsx (hunk 5 above)
Comment (alishaz-polymath) [reply pages/event-types/index.tsx]: Perhaps the instance didn't get updated but organizer does have the property username, as it is pulled from here:
https://github.com/calcom/cal.com/pull/1759/files#diff-758cb4a85756482b534d35a7806ed3bcdbd2775698e803a83a40773ba1c24582R134

**Review by zomars — APPROVED** [review]

**Pull request merged**